src/foam/dao/index/AltIndex.java

```diff
@@ -27,20 +27,30 @@ public void addIndex(Index i) {
     delegates_.add(i);
   }
 
+  public Object get(Object state, FObject obj) {
+    if ( state == null ) state = new Object[delegates_.size()];
```

Comment:
Make a helper method so that you can do:
Object[] s = myState(state);

Which casts to Object[] for you and allocates a new Object[] if required, so you don't have to duplicate this logic in each method and also have ((Object[])state)'s all throughout your code.